pkg/planner/core/rule_join_elimination.go

```diff
@@ -68,11 +68,28 @@ func (o *OuterJoinEliminator) tryToEliminateOuterJoin(p *logicalop.LogicalJoin,
 		}
 	}
 
-	// outer join elimination with duplicate agnostic aggregate functions
-	matched := ruleutil.IsColsAllFromOuterTable(aggCols, &outerUniqueIDs)
-	if matched {
-		appendOuterJoinEliminateAggregationTraceStep(p, outerPlan, aggCols, opt)
-		return outerPlan, true, nil
+	// Filter out constant columns from aggCols
+	nonConstCols := make([]*expression.Column, 0, len(aggCols))
+	for _, col := range aggCols {
+		if col.ID == 0 && !p.Schema().Contains(col) && col.VirtualExpr == nil {
```

Comment:
some check about removing the constant col is hard to judge here, it's not recommended to do that.

since aggCols are only allowed to appear in the outer table, how about use a function like `IsColsAllFromInnerTable ` here we pass inner table schema, if one col is not in inner table, we automatically think it comes from outer table, even it is generated from upper constant or ever. The logic will be much easier to do